Source/WebCore/page/FrameIdentifier.h

```diff
@@ -34,3 +34,33 @@ struct FrameIdentifierType;
 using FrameIdentifier = ProcessQualified<ObjectIdentifier<FrameIdentifierType>>;
 
 }
+
+namespace WTF {
+
+class FrameIdentiferStringTypeAdapter {
+public:
+    unsigned length() const { return lengthOfIntegerAsString(m_processIdentifier) + lengthOfIntegerAsString(m_objectIdentifier) + 1; }
+    bool is8Bit() const { return true; }
+    template<typename CharacterType> void writeTo(CharacterType* destination) const
+    {
+        writeIntegerToBuffer(m_processIdentifier, destination);
+        *(destination + lengthOfIntegerAsString(m_processIdentifier)) = '-';
```

Comment:
It would probably be worth storing the result of lengthOfIntegerAsString(m_processIdentifier) instead of calling it twice, though the compiler may be able to optimize it out.

Actually, this should probably go into ProcessIdentifier.h and use a StringTypeAdapter<T>